content/blog/2022-05-19-A_Type_and_Effect_System_for_Deterministic_Parallel_Java/index.md

```diff
@@ -0,0 +1,123 @@
++++
+title = "A Type and Effect System for Deterministic Parallel Java"
+[extra]
+bio = """
+  [Nikita Lazarev](https://www.nikita.tech/) is an PhD student at Cornell's ECE department;
+  Susan Garry is an undergraduate student at Cornell's CS department.
+"""
+[[extra.authors]]
+name = "Nikita Lazarev"
+link = "https://www.nikita.tech/"  # Links are optional.
+[[extra.authors]]
+name = " Susan Garry"
++++
+
+### Paper review
+
+["A Type and Effect System for Deterministic Parallel Java", Robert L. Bocchino, Vikram S. Adve, Danny Dig, Sarita V. Adve, Stephen Heumann, Rakesh Komuravelli, Jeffrey Overbey, Patrick Simmons, Hyojin Sung, and Mohsen Vakilian. OOPSLA 2009](http://dpj.cs.illinois.edu/DPJ/Publications_files/DPJ-OOPSLA-2009.pdf)
+
+### Introduction
+
+Multithreading is notoriously difficult to reason about and error-prone. This happens due to potential race conditions that can be arise when two threads try to access the same region in memory simultaneously. Traditional approaches using locks (e.g. mutexes) can cause a deadlock and are also difficult to reason about and debug.
```

Comment:
Add a comma after "e.g." https://capra.cs.cornell.edu/styleguide/#egie

(This occurs several places throughout the post, including with "i.e.")